src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -257,20 +266,33 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 			else {
 				diag('load() - created text editor model', this.resource, new Date());
 
-				this.createTextEditorModelPromise = this.createTextEditorModel(content.value, content.resource).then(() => {
-					this.createTextEditorModelPromise = null;
+				return this.backupFileService.doesTextFileHaveBackup(this.resource).then(backupExists => {
+					let getContentPromise: TPromise<IRawText>;
+					if (backupExists) {
+						const restoreResource = this.backupFileService.getBackupResource(this.resource);
+						const restoreOptions = { acceptTextOnly: true, encoding: 'utf-8' };
+						getContentPromise = this.textFileService.resolveTextContent(restoreResource, restoreOptions).then(content => content.value);
```

Comment:
@Tyriar if something goes wrong loading that backup, a user will never be able again to open the file. I think a safe guard should be added to fall back to the content from disk.